vendors/ti/boards/cc3220_launchpad/ports/secure_sockets/iot_secure_sockets.c

```diff
@@ -902,6 +902,203 @@ int32_t SOCKETS_Send( Socket_t xSocket,
 }
 /*-----------------------------------------------------------*/
 
+#if ( configPLATFORM_SOCKET_UDP_SUPPORT == 1 )
+    int32_t SOCKETS_Bind( Socket_t xSocket,
```

Comment:
Calling sendto or recvfrom make bind redundant, at least on other major platforms. Is that the intent of our port as well?

Reply:
I'm not sure if sendto and recvfrom make bind redundant. Bind allows API user to select the local port.

Reply:
To add  to above comment:  @dcgaws  is correct when UDP is limited to only client software. bind() is not explicitly needed, since a client does not have to listen to a specific local port. sendto will auto bind to a random port. If we ever want to support server like functionality , i.e., listen to a specific port, then bind is required.

Reply:
I think the question here is if an application can call `SOCKETS_RecvFrom()` or `Sockets_SendTo()` in this port without calling `SOCKETS_Bind()` first.  I suspect that the answer is yes if TI's implementation follows the typical behavior, but it would be good to have tests to show that this is the case.

Reply:
Yes there should be a test to address that. I'm only confident that TI does bind if not bound before a sendTo, since there were tests that were testing other aspect of sendTo without bound. This bring up a question of how do we test recvFrom without sendTo since we need to send data to echo sever before we can receive anything.